src/read/io/locator.rs

```diff
@@ -57,7 +57,7 @@ where
         let read = reader.read(&mut buffer).await?;
 
         if let Some(match_index) = reverse_search_buffer(&buffer[..read], signature) {
-            return Ok(position + ((match_index + 1) - SIGNATURE_LENGTH) as u64);
```

Comment:
I'm pretty confident that negating the signature length is required here as the `reverse_search_buffer()` provides an index to the last byte of the signature (since it matches in reverse, so the first occurrence is actually the last byte).

Ie. [50, 4b, 05, 06] would return an index of 3, so then we do (3+1) - 4 = 0.

The (albeit incomplete) tests do seem to catch this: https://gyazo.com/4b7fda42de81d8007b32a83a9bfa5978